src/test/java/BaseTest.java

```diff
@@ -3,38 +3,91 @@
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
 import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
 import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.annotations.*;
 
+import java.net.MalformedURLException;
+import java.net.URI;
 import java.time.Duration;
 import java.util.UUID;
 
 
 public class BaseTest {
-    public static WebDriver driver = null;
+   //remember if declaring non-primitive data types, must assign values-strings, arrays, classes
+    public static WebDriver driver;
     public static String url = null;
 
+    public static WebDriverWait wait = null;
+
+    public static Actions action;
```

Comment:
I recommend: make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

```
    protected Actions actions;
    protected WebDriverWait wait;
    protected WebDriver driver;
```